components/relevancy/src/lib.rs

```diff
@@ -159,8 +161,76 @@ impl RelevancyStore {
     /// its likelihood of a negative outcome.
     #[handle_error(Error)]
     pub fn bandit_update(&self, bandit: String, arm: String, selected: bool) -> ApiResult<()> {
+        let mut cache = self.cache.lock();
+
+        cache
+            .clear(&bandit, &arm)
+            .expect("failed to clear cached distribution");
+
```

Comment:
I was thinking about this case yesterday and I think we can do better here.  If the bandit/arm is cached, then rather than clear the entry we can update the entry based on the `selected` value.  I'm not sure if this is worth the complexity or not though.

Reply:
yeah this makes sense. if we're updating the cache with the new values, there's no need for the clear method then. If we update after writing to the DB, is that enough of a fail safe? What happens if the DB write fails?

Reply:
I think it's safe as long as the cache is updated after the DB operation completes.  If it works, then I think you can safely update the cache.  If not, then the transaction should be rolled back and it's safe to keep the older cached value.

Reply:
after some thought, I think clearing it and forcing a re-fetch is the best way to handle this. If not we'd have to make another call to retrieve the current values from the cache before updating, or make changes to the DB method to return the new values. It becomes a little complex to keep the cache up to date & consistent in cases of failures in the DB operation. We're plan on updating this logic to allow for bulk updates so maybe we leave it for now and revisit?